### ISSUE TITLE
Zypper example does not build

### ISSUE THREAD

**Issue by ipelupessy** [singularityhub/singularityhub.github.io#123]
The vanilla zypper example doesn't build on singularity hub

## Link to Container Collection Log, Build, or Collection (in that order)

https://www.singularity-hub.org/containers/2421/log

### Behavior when Building Locally

works.

### Error on Singularity Hub

Valid metadata not found at specified URL

### What do you think is going on?

not sure. maybe the GPG keys are not imported automatically?

 if I remove the OSVersion keyword and put it in by hand, the following error results:

 error: cannot open Packages index using db5 - Read-only file system (30)

**Comment by vsoch** [comment]
@cclerget is this a limitation of secure build?

**Comment by vsoch** [comment]
hey @ipelupessy ! I've just quietly added the new builder (with Singularity 2.5) as an option, and I think it would be worth a go to see if your container builds! To change your builder:

 - navigate to your Collection
 - click on "Settings"
 - under the Builder settings tag, change the cloud builder to `singularity-builder-2-5-100gb`

![image](https://user-images.githubusercontent.com/814322/39400175-4f814304-4af9-11e8-9f55-043ea9b45cd7.png)

And then commit and push again to trigger of course, and report back to me! If it's an issue with the secure build image then you can confirm by testing locally with the secure build branch, and I can give you instructions for that.

**Comment by ipelupessy** [comment]
@vsoch:

thanks!
for the 2.5 builder, I think the zypper install is broken/ misconfigured (maybe libzypp is not installed?):
```
/usr/bin/zypper: error while loading shared libraries: libzypp.so.1503: cannot open shared object file: No such file or directory
```
(https://www.singularity-hub.org/containers/2421/log )

**Comment by vsoch** [comment]
Yes so we need to ask @cclerget about this.

**Comment by vsoch** [comment]
I'm going to launch a build and see if I can debug this guy, is it the case that zypper is a different install from it's libraries (or is there a dev?) Any hints you can provide would be greatly helpful!

**Comment by vsoch** [comment]
okay, testing with adding:

```bash
sudo apt-get install -y libzypp libzypp-dev
```

**Comment by vsoch** [comment]
hey @ipelupessy this fix did not work, I am getting the same issue with those libraries:

```
sudo singularity build --isolated test.simg Singularity 
Using container recipe deffile: Singularity
Building into existing container: /root/build
Using container recipe deffile: Singularity
Sanitizing environment
Adding base Singularity environment to container
Found Zypper at: /usr/bin/zypper
/usr/bin/zypper: error while loading shared libraries: libzypp.so.1503: cannot open shared object file: No such file or directory
/usr/bin/zypper: error while loading shared libraries: libzypp.so.1503: cannot open shared object file: No such file or directory
/usr/bin/zypper: error while loading shared libraries: libzypp.so.1503: cannot open shared object file: No such file or directory
ERROR: Bootstrap failed... exiting
ABORT: Aborting with RETVAL=255
Cleaning up...
Cleaning up...
```
I will tell you how to figure this out, and you can tell me and I'll update the image with the fix. What you first need to do is install Singularity and compile the secure build feature:

```
git clone -b feature-squashbuild-secbuild-2.5.0 https://github.com/vsoch/singularity.git
cd singularity
```
In this folder is a file called `secbuild.sh.in` and in that file is a Bootstrap recipe echoed that is the base image to run the secure build. It looks like this in `%post`

```
%post 
    export LC_LANG=C
    apt-get update -y
    apt-get install -y git gcc python make automake autoconf libtool
    apt-get install -y squashfs-tools curl uuid-dev libssl-dev libarchive-dev
    apt-get install -y debootstrap yum zypper libzypp libzypp-dev
    apt-get clean
    apt-get autoclean
    rm -rf /usr/local/libexec/singularity
    rm -rf /usr/local/lib/singularity
    cd $SECBUILD_SINGULARITY
    ./configure --disable-suid
    make clean
    make -j4
    make install
DEFFILE
```

See the line where zypper is installed? That's where I tried appending those last two libraries and got the same error. So what you want to do is edit that build recipe above with whatever you think is needed in the container for it to work. When you are done editing the file, you need to 1) compile secure build, and then 2) do it with the `--isolated` flag. Let's walk through those steps:

```
# I just finished editing the file and then...
./autogen.sh && ./configure --prefix=/usr/local && make && sudo make install && sudo make secbuildimg
```
The key is the end bit - the `sudo make secbuildimg`

You should then be able to build your recipe file with `--isolated`

```
sudo singularity build --isolated test.simg Singularity 
```
@cclerget will comment if he has insight to this bug, otherwise I believe this is the best way to debug , and test.

**Comment by vsoch** [comment]
@ipelupessy have you been able to test with the secure build? Please follow the instructions above and let me know the result, thanks. It looks like secure build isn't available for Singularity 3.0.0 so unfortunately we won't be updating the builders for it.

**Comment by vsoch** [comment]
No response, I don't think I can be of help without an example base where the build does work. I also just noticed that ubuntu 18.04 no longer has zypper. Closing, please open a new issue with a recipe if you'd like me to try something different.
